src/storage/txn/scheduler.rs

```diff
@@ -1214,82 +1066,51 @@ impl<E: Engine> Scheduler<E> {
     fn release_lock(&mut self, lock: &Lock, cid: u64) {
         let wakeup_list = self.latches.release(lock, cid);
         for wcid in wakeup_list {
-            self.lock_and_register_get_snapshot(wcid);
+            self.lock_and_get_snapshot(wcid);
         }
     }
 
     /// Tries to acquire all the necessary latches. If all the necessary latches are acquired,
     /// the method initiates a get snapshot operation for furthur processing.
-    fn lock_and_register_get_snapshot(&mut self, cid: u64) {
+    fn lock_and_get_snapshot(&mut self, cid: u64) {
         if self.acquire_lock(cid) {
-            let ctx = self.extract_context(cid).clone();
-            let group = self
-                .grouped_cmds
-                .as_mut()
-                .unwrap()
-                .entry(HashableContext(ctx))
-                .or_insert_with(Vec::new);
-            group.push(cid);
+            self.get_snapshot(cid);
         }
     }
 }
 
 impl<E: Engine> Runnable<Msg<E>> for Scheduler<E> {
-    fn run(&mut self, msg: Msg<E>) {
-        match msg {
-            Msg::Quit => {
-                self.shutdown();
-                return;
-            }
-            Msg::RawCmd { cmd, cb } => self.on_receive_new_cmd(cmd, cb),
-            Msg::RetryGetSnapshots(batch) => self.on_retry_get_snapshots(batch),
-            Msg::SnapshotFinished {
-                cids,
-                cb_ctx,
-                snapshot,
-            } => self.on_snapshot_finished(cids, cb_ctx, snapshot),
-            Msg::BatchSnapshotFinished { batch } => for (cids, cb_ctx, snapshot) in batch {
-                self.on_snapshot_finished(cids, cb_ctx, snapshot)
-            },
-            Msg::ReadFinished { cid, pr } => self.on_read_finished(cid, pr),
-            Msg::WritePrepareFinished {
-                cid,
-                ctx,
-                tag,
-                pr,
-                to_be_write,
-                rows,
-            } => self.on_write_prepare_finished(cid, ctx, tag, pr, to_be_write, rows),
-            Msg::WritePrepareFailed { cid, err } => self.on_write_prepare_failed(cid, err),
-            Msg::WriteFinished {
-                cid, pr, result, ..
-            } => self.on_write_finished(cid, pr, result),
-        }
+    fn run(&mut self, _: Msg<E>) {
+        panic!("Shouldn't call Scheduler::run directly");
```

Comment:
I still prefer implementing `run()` and call it in `run_batch()` which is consistent with other codes